### PR TITLE
Hashing objs

### PR DESCRIPTION
**Pull request by miguelyermo-heidelberg**
Implementation of a memory cache to speedup the load of the same .obj multiple times.

- Limitations:
  - Instead of hashing each object, its faster to use the path + filename as a hashing key. The limitation of this is, if several objects are geometrically identical but used with different file names, they will be all cached. I can't think of a reason why this could happen, but the limitation is there.
 - Notes:
   - The implementation is done so only 32 (by default) objects will be cached at the same time. If the cache is full, the last loaded from cache object will be deleted to make room to the new one to be cached. If you want this number to be another, or even to be defined by the user as an input parameter, let me know.
   - In the process of implementing the algorithm, two obscure bugs where encountered and solved, both related with the management of the memory when copying SceneParts and Triangles (way far the most time consuming task)
- Observations:
  - The way the load and transformations of the objs are implemented is very strange in my opinion. The classes which take the responsibility of loading objs and performing transformations are children of the same base class, and a new instance of these objects are created every time a loading or a transformation has to be done.

### CODE REVIEW

File: src/scene/Scene.cpp [hunk 1]
```diff
@@ -19,225 +18,208 @@ using namespace glm;
...
+  // #####   UPDATE PRIMITIVES ON FINISH LOADING   #####
+  UniformNoiseSource<double> uns(-1, 1);
+  for (Primitive *p : primitives) {
+    p->onFinishLoading(uns);
+  }
+
+  // ################ BEGIN Shift primitives to originWaypoint
+  // ##################
+
+  // Translate scene coordinates to originWaypoint (to prevent wasting of
```
Comment (lwiniwar) [inline src/scene/Scene.cpp:78]: what is "originWaypoint" referring to in this comment? the bbox min is moved to 0,0,0 in line 99ff.

**Review by lwiniwar — COMMENTED** [review]
good work with the time watcher, this is definitely a useful feature in the future!

File: src/scene/Scene.cpp (hunk 1 above)
Comment (miguelyermo-heidelberg) [reply src/scene/Scene.cpp:78]: Well... I don't know. That part appears as a new change in the commit due to a reformat of the file that I did. The comment first appeared in in the first commit you did to this repository. f27b518, on 20/01/21@16.07h - "Init commit to github". I did not touch that part of the code at all.

Comment (lwiniwar) [reply src/scene/Scene.cpp:78]: fair. Please delete the misleading comment :)

**Pull request merged**